src/libraries/System.Text.Encodings.Web/src/System/Text/Encodings/Web/TextEncoder.cs

```diff
@@ -898,11 +898,11 @@ public virtual unsafe int FindFirstCharacterToEncodeUtf8(ReadOnlySpan<byte> utf8
             }
         }
 
-        internal static unsafe bool TryCopyCharacters(char[] source, char* destination, int destinationLength, out int numberOfCharactersWritten)
+        internal static bool TryCopyCharacters(string source, Span<char> destination, out int numberOfCharactersWritten)
         {
-            Debug.Assert(source != null && destination != null && destinationLength >= 0);
+            Debug.Assert(source.Length > 0);
```

Comment:
Could throw NRE in debug builds when we'd rather it assert. Suggest using `Debug.Assert(!string.IsNullOrEmpty(source))`.

Reply:
Oops. This was left over when `char[] source` went to `ReadOnlySpan`, which couldn't be `null`. And then I changed it to `string`. I'll fix this.